laravel-app/app/Http/Controllers/Auth/RegisterController.php

```diff
@@ -46,39 +47,89 @@ public function __construct()
 
     /**
      * Show login form page.
-     *
-     * @return View
      */
-    public function showRegistrationForm()
+    public function showRegistrationForm(): view
     {
         return view('auth.register');
     }
 
+    /**
+     * Show resend page
+     */
+    public function resendPage(): view
+    {
+        return view('auth.resend');
+    }
+
     /**
      * Create a new user instance after a valid registration.
-     *
-     * @param  CreateUserRequest  $request
-     * @return View
      */
-    public function register(CreateUserRequest $request): View
+    public function register(CreateUserRequest $request): RedirectResponse
     {
         $validatedData = $request->validated();
 
         $user =  User::create([
             'email' => $validatedData['email'],
             'password' => Hash::make($validatedData['password']),
             'username' => $validatedData['username'],
+            'email_verification_code' => Str::random(40),
         ]);
 
         $user->information()->create([
             'first_name' => $validatedData['first_name'],
             'last_name' => $validatedData['last_name'],
         ]);
 
-        event(new Registered($user));
+        /** @var User $user */
+        Mail::to($validatedData['email'])->send(new EmailVerificationMail($user));
+        return redirect()->route('login')->with('success', 'Registration complete. Please check email');
+    }
+
+    /**
+     * Verifies email after registration
+     */
+    public function verifyEmail(string $verificationCode): RedirectResponse
+    {
+        $user = User::where('email_verification_code', $verificationCode)->first();
+
+        if (!$user) {
+            return redirect()->route('resend')->with('error', 'Verification link expired. Resend another one.');
+        }
+
+        if ($user->email_verified_at) {
+            return redirect()->route('resend')->with('error', 'Email already verified');
+        }
+
+        $user->update([
+            'email_verified_at' => \Carbon\Carbon::now(),
+        ]);
+
+        return redirect()->route('login')->with('success', 'Email successfully verified');
+    }
+
+    /**
+     * Resends email for verification
+     */
+    public function resendEmail(ResendEmailRequest $request): RedirectResponse
+    {
+        $email = $request->input('email-resend');
+        $user = User::where('email', $email)->first();
+
+        /** @var User $user */
```

Comment:
Why still need this? you are calling directly with the model right?

Reply:
Yes sir, will resolve it 🙇